src/App.tsx

```diff
@@ -7,33 +9,66 @@ function getRandomName(): string {
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const date = new Date();
-  const clockName = getRandomName();
+type State = {
+  hasClock: boolean;
+  clockName: string;
+};
+
+export class App extends Component<{}, State> {
+  state: Readonly<State> = {
+    hasClock: true,
+    clockName: getRandomName(),
+  };
+
+  timerId = 0;
+
+  componentDidMount() {
+    this.timerId = this.setTimer;
+    document.addEventListener('click', this.handlerClock);
+    document.addEventListener('contextmenu', this.handlerClock);
+  }
+
+  componentDidUpdate(_: Readonly<{}>, prevState: Readonly<State>) {
+    const prevName = prevState.clockName;
+    const currentName = this.state.clockName;
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    // ...
-  }, 1000);
+    if (prevName !== currentName) {
+      console.log(`Renamed from <${prevName}> to <${currentName}>`); // eslint-disable-line
+    }
+  }
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+  componentWillUnmount() {
+    clearInterval(this.timerId);
```

Comment:
componentWillUnmount() don't work for clearing Interval, because App doesn't close or reloaded. Think "clearInterval(this.timerId)" should be placed in "componentDidMount()" method when the state is changed to "hasClock: false"
<img width="337" alt="image" src="https://user-images.githubusercontent.com/106744410/183076216-3fd23cf0-2bff-40d8-a6ff-4fb0daaaa8b7.png">